src/Schema/Table.php

```diff
@@ -735,6 +736,10 @@ public function getForeignKeyColumns()
      */
     private function filterColumns(array $columnNames, bool $reverse = false): array
     {
+        $columnNames = array_map(function (string $columnName): string {
```

Comment:
Wouldn't it be enough to normalize the name during filtering? Why do we need a list of normalized names?

Reply:
You could normalize during the filtering, however, that would mean the sanitazion is done for every record, making it O(n), instead of doing it once where it is O(1). 

As said above, ideally we wouldn't require a list of sanitized names in the first place.

Reply:
It's O(n) in both cases. Please move within `array_filter()`.

As for normalization, getting rid of it is part of a bigger challenge (see https://github.com/doctrine/dbal/issues/4357). We can consider it a requirement for now.